File: 2/TRPP/Access_Control/Access_Control/main/src/main/EmployeeController.cpp
// src/main/java/com/example/demo/controller/EmployeeController.java
package com.example.demo.controller;
import com.example.demo.model.Employee;
import com.example.demo.service.EmployeeService;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import org.springframework.web.multipart.MultipartFile;
import java.util.List;
@RestController
@RequestMapping("/employees")
public class EmployeeController {
    private final EmployeeService employeeService;
    public EmployeeController(EmployeeService employeeService) {
        this.employeeService = employeeService;
    }
    @PostMapping("/add")
    public ResponseEntity<?> addEmployee(
            @RequestParam("name") String name,
            @RequestParam("age") int age,
            @RequestParam("idOfEnterprise") Long idOfEnterprise,
            @RequestParam(value = "file", required = false) String file,
            @RequestParam("fromCamera") boolean fromCamera) {
        Employee employee = new Employee();
        employee.setName(name);
        employee.setAge(age);
        employee.setIdOfEnterprise(idOfEnterprise);
        boolean success = employeeService.addEmployee(employee, file, fromCamera);
        return success
                ? ResponseEntity.ok("Employee added successfully.")
                : ResponseEntity.badRequest().body("Failed to add employee.");
    }

    @DeleteMapping("/remove/{id}")
    public ResponseEntity<?> removeEmployee(@PathVariable Long id) {
        boolean success = employeeService.removeEmployee(id);
        return success
                ? ResponseEntity.ok("Employee removed successfully.")
                : ResponseEntity.badRequest().body("Employee not found.");
    }
    @GetMapping("/poll")
    public ResponseEntity<?> poll() {
        boolean result = employeeService.poll();
        return ResponseEntity.ok(result);
    }
    @GetMapping("/getall")
    public ResponseEntity<List<Employee>> getAllEmployees() {
        List<Employee> employees = employeeService.getAllEmployees();
        return ResponseEntity.ok(employees);
    }
}

File: 2/TRPP/H/Access_Control/main/src/main/java/com/example/demo/controller/EmployeeController.java
package mock_server.controller;

import com.fasterxml.jackson.databind.ObjectMapper;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import mock_server.service.EmployeeService;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;
import java.util.Map;
import java.util.stream.Collectors;

@RestController
@RequestMapping("/structure")
  
public class EmployeeController {
    private final EmployeeService employeeService;

    public EmployeeController(EmployeeService employeeService) {
        this.employeeService = employeeService;
    }

    @PostMapping("/add_employee")
    public ResponseEntity<Map<String, Object>> addEmployee(
            @RequestParam("data") String data,
            @RequestParam("file") MultipartFile file
    ) {
        try {
            Map json;
            json = new ObjectMapper().readValue(data, Map.class);
      
            String uuid = (String) json.get("uuid");
            employeeService.addEmployee(uuid, file);
            return ResponseEntity.ok(Map.of(
                    "status", "success",
                    "message", "Employee added successfully.",
                    "uuid", uuid
            ));
        } catch (Exception e) {
            return ResponseEntity.badRequest().body(Map.of(
                    "status", "error",
                    "message", e.getMessage()
            ));
        }
    }

    @DeleteMapping("/remove_employee")
    public ResponseEntity<Map<String, Object>> removeEmployee(@RequestBody Map<String, String> body) {
        String uuid = body.get("uuid");
        boolean removed = employeeService.removeEmployee(uuid);
        if (removed) {
            return ResponseEntity.ok(Map.of(
                    "status", "success",
                    "message", "Employee removed successfully."
            ));
        }
        return ResponseEntity.badRequest().body(Map.of(
                "status", "error",
                "message", "Employee not found."
        ));
    }

    @PostMapping("/check_employee")
    public ResponseEntity<Map<String, Object>> checkEmployee(@RequestParam("file") MultipartFile file) {
        String uuid = employeeService.checkEmployee(file);
        if (uuid != null) {
            return ResponseEntity.ok(Map.of(
                    "status", "success",
                    "message", "Employee recognized.",
                    "uuid", uuid
            ));
        }
        return ResponseEntity.status(HttpStatus.NOT_FOUND).body(Map.of(
                "status", "error",
                "message", "Employee not recognized."
        ));
    }

    @GetMapping("/get_employees")
    public ResponseEntity<Map<String, Object>> getEmployees() {
        List<String> faces = employeeService.getEmployees();
        return ResponseEntity.ok(Map.of(
                "status", "success",
                "employees", faces.stream().map(uuid -> Map.of("uuid", uuid)).collect(Collectors.toList())
        ));
    }
}
